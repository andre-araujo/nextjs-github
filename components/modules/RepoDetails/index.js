import {
    number,
    shape,
    string,
} from 'prop-types';

import { Box } from 'grid-styled';

import {
    Description,
    Label,
    Title,
} from './styles';

const RepoDetails = ({
    data,
}) => {
    const infoToDisplay = [
        'full_name',
        'forks',
        'language',
        'name',
        'open_issues',
        'stargazers_count',
        'subscribers_count',
        'homepage',
        'html_url',
    ];

    return (
        <section>
            <Title>{data.name}</Title>
            {
                infoToDisplay
                    .filter(info => data[info])
                    .map(info => (
                        <Box my="2rem" key={info}>
                            <Label>{info.replace('_', ' ')}</Label>

                            <Description>
                                {
                                    (/http/).test(data[info]) ?
                                        <a
                                            target="_blank"
                                            href={data[info]}
                                        >
                                            {data[info]}
                                        </a> :
                                        data[info]
                                }

                            </Description>
                        </Box>
                    ))
            }
        </section>
    );
};


RepoDetails.defaultProps = {
    data: {},
};

RepoDetails.propTypes = {
    data: shape({
        forks: number,
        full_name: string,
        homepage: string,
        html_url: string,
        language: string,
        name: string,
        open_issues: number,
        stargazers_count: number,
        subscribers_count: number,
    }).isRequired,
};

export default RepoDetails;
