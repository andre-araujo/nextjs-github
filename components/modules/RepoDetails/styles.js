import styled from 'styled-components';

export const Title = styled.h2`
    border-bottom: 1px solid #888;
    font-family: AvantGardeDemi;
    font-size: 2rem;
    line-height: 1;
    margin-bottom: .3em;
    padding-bottom: .3em;
    text-transform: uppercase;
`;

export const Label = styled.h3`
    font-family: AvantGardeMedium;
    font-size: 1.5rem;
    margin-bottom: .5rem;
    text-transform: capitalize;
`;

export const Description = styled.p`
    font-family: AvantGardeLight;
    font-size: 1.2rem;
    margin-right: .5rem;
`;
