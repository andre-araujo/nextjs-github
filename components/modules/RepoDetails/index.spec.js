import React from 'react';
import { mount } from 'enzyme';

import RepoDetails from './index';

describe('RepoDetails component', () => {
    it('should match snapshot with only full_name content', () => {
        expect(mount(
            <RepoDetails
                data={{
                    full_name: 'full_name',
                }}
            />,
        )).toMatchSnapshot();
    });

    it('should match snapshot with all repo data', () => {
        expect(mount(
            <RepoDetails
                data={{
                    full_name: 'full_name',
                    forks: 2,
                    language: 'language',
                    name: 'name',
                    open_issues: 15,
                    stargazers_count: 515,
                    subscribers_count: 151,
                    homepage: 'homepage',
                    html_url: 'http',
                }}
            />,
        )).toMatchSnapshot();
    });
});
