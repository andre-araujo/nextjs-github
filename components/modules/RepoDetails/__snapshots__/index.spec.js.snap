// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoDetails component should match snapshot with all repo data 1`] = `
<section>
  <h2
    className="sc-bxivhb jTgEYt"
  >
    name
  </h2>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      full name
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      full_name
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      forks
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      2
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      language
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      language
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      name
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      name
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      open issues
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      15
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      stargazers count
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      515
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      subscribers count
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      151
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      homepage
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      homepage
    </p>
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      html url
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      <a
        href="http"
        target="_blank"
      >
        http
      </a>
    </p>
  </div>
</section>
`;

exports[`RepoDetails component should match snapshot with only full_name content 1`] = `
<section>
  <h2
    className="sc-bxivhb jTgEYt"
  />
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <h3
      className="sc-ifAKCX kgARlw"
    >
      full name
    </h3>
    <p
      className="sc-EHOje cheXru"
    >
      full_name
    </p>
  </div>
</section>
`;
